lib/cisco_node_utils/yang.rb

```diff
@@ -0,0 +1,156 @@
+# June 2016, Charles Burkett
+#
+# Copyright (c) 2015-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'json'
+
+module Cisco
+  # class Cisco::Yang
+  # Utility class mainly containing methods to compare YANG configurations
+  # in order to determine if the running config is in-sync with the
+  # desired config.
+  class Yang
+    # Is the specified yang string empty?
+    def self.empty?(yang)
+      !yang || yang.empty?
+    end
+
+    # Given a current and target YANG configuration, returns true if
+    # the configurations are in-sync, relative to a "merge_config" action
+    def self.insync_for_merge?(target, current)
```

Comment:
Do you actually need two different methods here for `:merge` and `:replace` or could you collapse them into a single method like `self.insync_for_action?(target, current, action).  Not a bit deal but saves a few lines?